### ISSUE TITLE
Matrix/ Discord

### ISSUE THREAD

**Issue by Shoalsteed** [Shoalsteed/Events#2]

**Comment by marek22k** [comment]
About security, access, and moderation:
Regarding security, I know the following: The bot / bridge is operated by http://t2bot.io. This is free or open source. You could do this yourself - if you have knowledge, resources and time. Unfortunately we cannot encrypt the rooms because the bot will then no longer work.
However, since you need access to the room from the server where the matrix room is located, it is still private - similar to Discord: The chat is not encrypted. However, the platform does not allow everyone to join.
Regarding the moderation - I think - Matrix is very nicely designed. A user can be assigned an authorization from 0 to 100 (0 = can only talk; 100 = can do anything). What the individual authorization levels are allowed to do can be changed in the respective room settings. A moderator has level 50 and an admin level 100. A user is usually given level 0.
To get access to the Matrix "Space" I2P or to the rooms you need an invitation.

**Comment by Shoalsteed** [comment]
My questions would be:
- how would be handle invites
- what makes Matrix different ( ie is it better ?)
- with its moderation capacities, can we ensure that the issues that occur on irc do not happen or get transferred to Matrix/ Discord? I am talking about trolling, abuse, bots, etc that have resulted in unproductive and unwelcoming spaces.

**Comment by marek22k** [comment]
Matrix itself wants to be compatible with as many other platforms as possible.
![matrix-bridges](https://user-images.githubusercontent.com/46723399/143309502-7acee28c-4d3e-4fd1-b960-733ba34dc6c6.png)

**Comment by marek22k** [comment]
>> how would be handle invites
Depending on the setting, admin, mods and even users can send out invitations. A thought game: It would be possible to give "particularly trustworthy" users level 10 and to specify in the settings that all users from level 10 are allowed to create invitations.
>> what makes Matrix different ( ie is it better ?)
That's a pretty simple question. However, there are many different aspects to the answer. For me, two aspects are special:
1. It's not centralized. Everyone can open their own matrix server and create rooms there. The rooms are then synchronized to the other server (depending on which server members are there).
2. It's open source and free.
>> with its moderation capacities, can we
Well, an invitation-only principle, for example, would help against this (as we also have it on Discord). Furthermore admins and mods have the possibility to moderate manually.

**Comment by Shoalsteed** [comment]
I am in favour of adding on / using anything that will allow more people to engage in productive conversations about the project. As long as we decide who and how we will care for it!

Lets get some ideas from idk as well.

**Comment by eyedeekay** [comment]
I like the idea of using Matrix more within I2P, and would like to even consider running our own homeservers inside of I2P with Dendrite. I've got a WIP branch of Dendrite where I am adding I2P support over here https://github.com/eyedeekay/dendrite but it's not quite ready yet. I think Matrix manages to have most of the good things about IRC while also solving some of the problems of IRC, and it's overt effort to build bridges to other networks is appealing to me too. +1 to matrix from me.

**Comment by Shoalsteed** [comment]
Other things:
- Do we need some sort of "rules" 
- would we want to make more rooms?
- would we want to make it official and promote it?

**Comment by marek22k** [comment]
> I am in favour of adding on / using anything that will allow more people to engage in productive conversations about the project. As long as we decide who and how we will care for it!
> 
> Lets get some ideas from idk as well.

This statement could be a bit pessimistic, but it gives security: If everything goes wrong and something "bad" happens, you just need Discord to remove the bot's permission and to kick it out and the whole bridge would be hard disabled.

**Comment by marek22k** [comment]
![Screenshot at 2021-11-24 21-48-00](https://user-images.githubusercontent.com/46723399/143310841-07226785-3bf6-440d-b262-2cf3e26cb3b9.png)

**Comment by Shoalsteed** [comment]
> > I am in favour of adding on / using anything that will allow more people to engage in productive conversations about the project. As long as we decide who and how we will care for it!
> > Lets get some ideas from idk as well.
> 
> This statement could be a bit pessimistic, but it gives security: If everything goes wrong and something "bad" happens, you just need Discord to remove the bot's permission and to kick it out and the whole bridge would be hard disabled.

Yeah, I realize it does sound that way, I have been witness to some pretty poor actions by others in the past that have informed some of my judgement.

**Comment by marek22k** [comment]
I hope I'm not too annoyed by the question: when should this be decided?
Personally, I would be interested in it being decided soon, because then I wouldn't have to bother with Discord anymore.

**Comment by eyedeekay** [comment]
To be honest I would be quite glad to be rid of Discord as well for the same reasons marek22k would be, I do not like that Discord has greater control of that communication channel than the I2P team in a strictly technical sense. I will always choose Software Libre over anything else for my personal use, and recommending anything else to anyone else is not something I ever do lightly. That said I do support going "where the people are" to talk to them so I recognize that utility in Discord. But the thing is that Matrix has that power too, and someday soon, it will have that power in greater quantities than Discord, with greater flexibility than Discord, and similar moderation capability to Discord, abd it can be ours instead of Discord's, AND it can still talk to Discord. That's winning all around to me, as long as there is a Matrix channel moderator as well as a Discord channel moderator.

**Comment by marek22k** [comment]
I just remember: Regarding the invitation system in Matrix, it differs from Discord. Instead of sending an invitation link, the user is invited directly in Matrix. This also prevents misuse of the invitation link.

**Comment by eyedeekay** [comment]
There are just so many benefits to Matrix over both IRC and Discord, IMO this should be a thing we do on a trial basis as soon as we can run a homeserver in I2P, and if it works we just keep it going.

**Comment by Shoalsteed** [comment]
> There are just so many benefits to Matrix over both IRC and Discord, IMO this should be a thing we do on a trial basis as soon as we can run a homeserver in I2P, and if it works we just keep it going.

Should we try with the way @marek22k is suggesting for now?

**Comment by Shoalsteed** [comment]
what are the steps I need to do ?

**Comment by eyedeekay** [comment]
For now the t2bot way should be fine, I think I'll need to defer to marek22k's knowledge for what you will need to do Shoalsteed

**Comment by marek22k** [comment]
https://discordapp.com/api/oauth2/authorize?client_id=309408702530846730&scope=bot&permissions=607251456
As I said before, the Discord server owner (I think that's Sadie) would have to add the bot and give it appropriate permissions.
"Verify that the bot is allowed to see the channel on Discord, and that it has permission to manage webhooks and send messages." Quote from troubleshooting

**Comment by Shoalsteed** [comment]
Okay, if we want to do this, I will set it up tomorrow!

**Comment by marek22k** [comment]
Did you add the bot to Discord?

**Comment by Shoalsteed** [comment]
> Did you add the bot to Discord?

I ended up getting pretty deep into the docs and the information architecture work yesterday so did not. On my list! Sorry for that.

**Comment by marek22k** [comment]
Have you added the bot now? That should go very quickly. Just invite the bot, give it permissions, and that's it.

**Comment by marek22k** [comment]
@Shoalsteed Ping. Did you add the bot?

**Comment by Shoalsteed** [comment]
I have not added it yet.

There are several things that have come up that need my attention that I did not foresee when saying "iI will do that tomorrow"

I would also like to have some sort of plan in place for having the Matrix option.

**Comment by marek22k** [comment]
It's been almost a month since we decided to build a bridge. I would like to continue to participate. However, this is only possible to a very limited extent, as a lot of communication takes place via Discord. The problem would be solved with the bridge. This is also one of the reasons why I am pushing it so hard. I think @Shoalsteed  if you give @eyedeekay the appropriate permissions on Discord he can add the bot too. Either way, I watched the tutorial on it and found that adding the bot is not a big deal.

**Comment by eyedeekay** [comment]
I asked @marek22k to make us some diagrams of how intra-platform modding may/should work to help move this discussion along when there is a chance to act on it.

![image](https://user-images.githubusercontent.com/8733713/148128835-98378af1-290a-43aa-8ea9-16983c341bb6.png)
![image](https://user-images.githubusercontent.com/8733713/148129041-9474722a-2055-4c87-a729-1394defa8496.png)

**Comment by Shoalsteed** [comment]
Happy to look at this next week, I am catching up this week.

**Comment by Shoalsteed** [comment]
@marek22k I added the bot
